### PR TITLE
docs(Request): clarify auto_parse_qs_csv behavior for splitting

### LINKED ISSUE #1597: RequestOptions.auto_parse_qs_csv documentation is misleading for the True case
[`uri.parse_query_string`](https://falcon.readthedocs.io/en/stable/api/util.html#falcon.uri.parse_query_string) supports splitting parameter values on comma, however, only the first occurrence is actually split:
```python
>>> uri.parse_query_string('interesting&stuff=1,2&key=value&stuff=3,4,5', csv=True)
{'stuff': ['1', '2', '3,4,5'], 'key': 'value'}
```

This is a known and documented limitation of `uri.parse_query_string`:
> The two different ways of specifying lists may not be mixed in a single query string for the same parameter.

However, documentation for [`RequestOptions.auto_parse_qs_csv`](https://falcon.readthedocs.io/en/stable/api/api.html#falcon.RequestOptions.auto_parse_qs_csv) mentions neither this limitation nor the fact that `uri.parse_query_string` is used for parsing. Moreover, it provides a misleading (although the example values are technically correct since only the first occurrence of `t` has commas) example:
> When `auto_parse_qs_csv` is set to ``True``, the query string value
> is also split on non-percent-encoded commas and these items
> are added to the final list (i.e. ``/?t=1,2,3&t=4``
> becomes ``['1', '2', '3', '4']``).

### PR DESCRIPTION
**Pull request by danilito19**
# Summary of Changes

Clarify documentation for `auto_parse_qs_csv` to indicate that only the first set of values for a given key are split by comma and subsequent values are left as is.

# Related Issues

Closes #1597

# Pull Request Checklist

This is just a reminder about the most common mistakes.  Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [ ] Applied changes to both WSGI and ASGI code paths and interfaces (where applicable).
- [ ] Added **tests** for changed code.
- [ ] Prefixed code comments with GitHub nick and an appropriate prefix.
- [ ] Coding style is consistent with the rest of the framework.
- [ ] Updated **documentation** for changed code.
    - [ ] Added docstrings for any new classes, functions, or modules.
    - [ ] Updated docstrings for any modifications to existing code.
    - [ ] Updated both WSGI and ASGI docs (where applicable).
    - [ ] Added references to new classes, functions, or modules to the relevant RST file under `docs/`.
    - [ ] Updated all relevant supporting documentation files under `docs/`.
    - [ ] A copyright notice is included at the top of any new modules (using your own name or the name of your organization).
    - [ ] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/usage/restructuredtext/directives.html?highlight=versionadded#directive-versionadded).
- [ ] Changes (and possible deprecations) have [towncrier](https://towncrier.readthedocs.io/en/actual-freaking-docs/index.html) news fragments under `docs/_newsfragments/`, with the file name format `{issue_number}.{fragment_type}.rst`. (Run `towncrier --draft` to ensure it renders correctly.)

If you have *any* questions to *any* of the points above, just **submit and ask**! This checklist is here to *help* you, not to deter you from contributing!

*PR template inspired by the attrs project.*

### CODE REVIEW

File: falcon/request.py [hunk 1]
```diff
@@ -1886,12 +1886,17 @@ class RequestOptions:
...
             When `auto_parse_qs_csv` is set to ``True``, the query string value
             is also split on non-percent-encoded commas and these items
-            are added to the final list (i.e. ``/?t=1,2,3&t=4``
+            are added to the final list (i.e. ``t=1,2,3&t=4``
             becomes ``['1', '2', '3', '4']``).
 
+            Warning:
+                When `auto_parse_qs_csv` is set to either ``True`` or ``False``,
+                only the first set of values for the same key are are split.
+                For example, ``t=1,2&t=3,4`` becomes ``['1', '2', '3,4'``.
```
Comment (vytas7) [inline falcon/request.py:1898]: Missing closing `]`

File: falcon/request.py [hunk 2]
```diff
@@ -1886,12 +1886,17 @@ class RequestOptions:
...
-            query string (i.e. ``/?t=1,2,3&t=4`` becomes ``['1,2,3', '4']``).
+            query string (i.e. ``t=1,2,3&t=4`` becomes ``['1,2,3', '4']``).
             When `auto_parse_qs_csv` is set to ``True``, the query string value
             is also split on non-percent-encoded commas and these items
-            are added to the final list (i.e. ``/?t=1,2,3&t=4``
+            are added to the final list (i.e. ``t=1,2,3&t=4``
             becomes ``['1', '2', '3', '4']``).
 
+            Warning:
+                When `auto_parse_qs_csv` is set to either ``True`` or ``False``,
```
Comment (vytas7) [inline falcon/request.py:1896]: I'm probably totally misunderstanding this line, but could you clarify what you mean with
> When `auto_parse_qs_csv` is set to either ``True`` or ``False``

As this option is designated as boolean, I expect it to be always set to either ``True`` or ``False`` :thinking:

**Review by vytas7 — CHANGES_REQUESTED** [review]
Thanks for working on this!

File: falcon/request.py (hunk 2 above)
Comment (danilito19) [reply falcon/request.py:1896]: Yes exactly! What I'm saying is it doesn't matter what `auto_parse_qs_csv` is set to (either True or False), but either way this warning applies. Does that make sense?

**Comment by danilito19** [comment]
@vytas7 I'll update the missing `]` but question before that... I see you've merged in master and now there are 2 extra commits in this PR. What git strategy do you prefer I follow? I'm used to always rebase, so those 2 extra commits would not show up on this PR. Instead, I would've rebased on the latest master. That keeps a cleaner git history.

**Comment by vytas7** [comment]
@danilito19 the internal PR commit hierarchy does not matter, as we "Squash and merge" 'em.

**Comment by danilito19** [comment]
Oh I see. This isn't my preferred way as I think keeping commit history is important, but I assume in an open source project with such a variety of authors and different commit styles (some I assume with not the best commit hygiene) this is probably for the best!

**Comment by vytas7** [comment]
@danilito19 sorry for the late answer to the squashing discussion. We should probably ask @kgriffs as this had already been that way when I started contributing to the project :slightly_smiling_face: 

But yeah, I do think it serves us well, given that various contributors have completely different working styles, ranging from force-push of a huge PR to a huge number of iterative twoliners.

File: falcon/request.py (hunk 2 above)
Comment (vytas7) [reply falcon/request.py:1896]: Maybe it's just me, but no, admittedly I'm still not following :imp:.

If `auto_parse_qs_csv` is set to the default (`False`) value, how does the following apply? :arrow_lower_left: 
> only the first set of values for the same key are are split.

To my understanding, if `auto_parse_qs_csv` is set to `False`, no values are split (at least in this context).

**Review by vytas7 — CHANGES_REQUESTED** [review]

File: falcon/request.py (hunk 2 above)
Comment (CaselIT) [reply falcon/request.py:1896]: agree

**Review by CaselIT — CHANGES_REQUESTED** [review]

**Comment by vytas7** [comment]
@danilito19 just checking, will you have an opportunity to bring this over the finish line this month?
Being a documentation change, this is not blocking the impending beta release, but it would be great to include this improvement in 3.0.

**Comment by danilito19** [comment]
Thanks so much for reminding me of this! I'm right in the middle of a
cross-country move, so I can't for good conscience dedicate time. Please
feel free to take this over so it gets in!

**Comment by vytas7** [comment]
No worries @danilito19; first things first. And thank you for your contributions so far, notably that docstrings conventions PR!

Hoping to see you back on board next year :wave:

File: falcon/request.py (hunk 2 above)
Comment (kgriffs) [reply falcon/request.py:1896]: I was able to confirm by adding a unit test that this only occurs when the option is enabled.

**Comment by kgriffs** [comment]
Thoughts on simply fixing the issue? See: #1833

**Comment by kgriffs** [comment]
Superseded by #1833

**Pull request closed without merging**